set.mm

```diff
@@ -424867,18 +424868,22 @@ involves a class difference ("dif") of a subset ("ss"), and thus is
        </li>
 
        <li><b>Suffixes.</b>
-       Suffixes are used to indicate the form of a theorem (see above).
-       Additionally, we sometimes suffix with "v" the label of a theorem
-       eliminating a hypothesis such as ` F/ x ph ` in ~ 19.21 via the use of
-       disjoint variable conditions combined with ~ nfv .  If two (or three)
-       such hypotheses are eliminated, the suffix "vv" resp. "vvv" is used,
-       e.g. ~ exlimivv .
+       Suffixes are used to indicate the form of a theorem (inference,
+       deduction, or closed form, see above).
+       Additionally, we sometimes suffix with "v" the label of a theorem adding
+       a disjoint variable condition, as in ~ 19.21v versus ~ 19.21 .  This
+       often permits to prove the result using fewer axioms, and/or to
+       eliminate a nonfreeness hypothesis (such as ` F/ x ph ` in ~ 19.21 ).
+       If no constraint is put on axiom use, then the v-version can be proved
+       from the original theorem using ~ nfv .  If two (resp. three) such
+       disjoint variable conditions are added, then the suffix "vv" (resp.
+       "vvv") is used, e.g., ~ exlimivv .
        Conversely, we sometimes suffix with "f" the label of a theorem
```

Comment:
I'm still not super happy about this. Following this convention proposal the name [chvarfv](https://us.metamath.org/mpeuni/chvarfv.html) would suggest the addition of a non-free hypothesis to remove a $d condition and the addition of another $d condition to drop axiom usage to [chvar](https://us.metamath.org/mpeuni/chvar.html), but in reality [chvarfv](https://us.metamath.org/mpeuni/chvarfv.html) and [chvar](https://us.metamath.org/mpeuni/chvar.html) differ by a $d condition only (their hypothesis are identical). So following your conventions update the appropriate name for `chvarfv` would be `chvarv`, but that's already taken so it can't be that.

To be clear here is how it is currently in the webpage:
* [chvar](https://us.metamath.org/mpeuni/chvar.html) is the "base version".
* [chvarv](https://us.metamath.org/mpeuni/chvarv.html) differs from the base version by replacing a non-free hypothesis with a <span>$</span>d condition between $\Psi$ and x.
* [chvarfv](https://us.metamath.org/mpeuni/chvarfv.html) differs from the base version by adding a $d condition between x and y (without replacing anything).

I think with this new proposal the `f` and `v` suffixes would interact between each other in a convoluted way. The current conventions are more clear about this: the v and f suffixes are simply the inverse of each other, one adds a $d condition to remove a non-free hyp while the other adds a non-free hyp to remove a $d, I can easily recognise what to expect in the statement just by looking at the label.

Reply:
No labeling scheme can be completely systematic and consistent if we don't want labels to be basically as long as the statements themselves. That's why many of these conventions use the adverb "sometimes".

In a given theorem ~xxx, there can be multiple natural DV conditions to add, so ~xxxv won't be unambiguous. Similarly, one may consider multiple natural nonfreeness conditions, so ~xxxf won't be unambiguous either.

The practice of suffixing a "v" when adding a DV condition is simple, and is followed by many existing theorems. Again, one does not necessarily add a DV condition in order to remove a nonfreeness hypothesis, but sometimes it can be in order to use fewer axioms.

I propose we stay like this at least for now.

Reply:
I agree with `No labeling scheme can be completely systematic` especially in a context like metamath where we want labels to be short and memorable. I don't think I have much of an opinion about the proposed change here - it strikes me as maybe a bit longer than it needs to be and maybe gets too much into details which don't need to be covered here, but I'm not sure it would really cause problems either.